### PR TITLE
Chore: enable e2e tests

### PR DESCRIPTION
**Pull request by katspaugh**
This PR makes Cypress run on each PR.

## How it works

* It builds static HTML files
* Serves them locally in the action container
* Runs cypress on localhost:8080

Because we use dynamic URLs, the server will actually respond with a 404 for URLs with a safe address. Although the page still renders. To mitigate this, in each test we should do this in the visit call:

```
cy.visit(`/${SAFE}/home`, { failOnStatusCode: false })
```

This tells Cypress to ignore 404s.

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -7,21 +7,39 @@ jobs:
...
+          yarn serve &
+
+      - uses: cypress-io/github-action@v4
         with:
-          spec: cypress/integration/smoke/*.cy.js
+          spec: cypress/e2e/smoke/*.cy.js
           browser: chrome
           record: false
-          config: baseUrl=https://${{ steps.extract_branch.outputs.branch }}.web-core.pages.dev
+          config: baseUrl=http://localhost:8080
```
Comment (yagopv) [inline .github/workflows/e2e.yml:42]: localhost:8080 ? dev stuff ?

**Review by yagopv — COMMENTED** [review]
There is a recorded video in the PR I guess you don't want to push

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (katspaugh) [reply .github/workflows/e2e.yml:42]: No, that's legit. Check the PR description.

**Comment by katspaugh** [comment]
Removed the video, thanks @yagopv.

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (yagopv) [reply .github/workflows/e2e.yml:42]: Ok!

**Review by yagopv — APPROVED** [review]

**Pull request merged**